netket/legacy/stats/mpi_stats.py

```diff
@@ -0,0 +1,137 @@
+import numpy as _np
+from ._sum_inplace import sum_inplace as _sum_inplace
+
+from netket.utils import (
+    mpi_available as _mpi_available,
+    n_nodes as _n_nodes,
+    MPI_py_comm as MPI_comm,
+)
+
+if _mpi_available:
+    from netket.utils import MPI
+
+
+def subtract_mean(x, axis=None):
+    """
+    Subtracts the mean of the input array over all but the last dimension
+    and over all MPI processes from each entry.
+
+    Args:
+        x: Input array
+        axis: Axis or axes along which the means are computed. The default (None) is to
+              compute the mean of the flattened array.
+
+    Returns:
+        The resulting array.
+
+    """
+    x_mean = mean(x, axis=axis)
+    x -= x_mean
+
+    return x
+
+
+def mean(a, axis=None, keepdims: bool = False):
+    """
+    Compute the arithmetic mean along the specified axis and over MPI processes.
+
+    Returns the average of the array elements. The average is taken over the flattened array by default,
+    otherwise over the specified axis. float64 intermediate and return values are used for integer inputs.
+
+    Args:
+        a: The input array
+        axis: Axis or axes along which the means are computed. The default (None) is to
+              compute the mean of the flattened array.
+        keepdims: If True the output array will have the same number of dimensions as the input,
+              with the reduced axes having length 1. (default=False)
+
+    Returns:
+        The array with reduced dimensions defined by axis.
+
+    """
+    out = a.mean(axis=axis, keepdims=keepdims)
+
+    out = _sum_inplace(out)
+    out /= _n_nodes
+
+    return out
+
+
+def sum(a, axis=None, out=None, keepdims: bool = False):
+    """
+    Compute the arithmetic mean along the specified axis and over MPI processes.
```

Comment:
```suggestion
    Compute the sum along the specified axis and over MPI processes.
```

Reply:
same in the non-legacy mpi_stats

(which is already with jax. I got confused finding it since github can't seem to handle such large PR's very well; sorry)